src/Cli.Tests/EndToEndTests.cs

```diff
@@ -167,6 +167,35 @@ public void TestAddEntity()
         Assert.AreEqual(EntityActionOperation.All, entity.Permissions[0].Actions[0].Action);
     }
 
+    /// <summary>
+    /// Test to verify telemetry details are added to the config.
+    /// </summary>
+    [TestMethod]
+    public void TestAddTelemetry()
+    {
+        string[] initArgs = { "init", "-c", TEST_RUNTIME_CONFIG_FILE, "--host-mode", "development", "--database-type",
+            "mssql", "--connection-string", TEST_ENV_CONN_STRING };
+        Program.Execute(initArgs, _cliLogger!, _fileSystem!, _runtimeConfigLoader!);
+
+        Assert.IsTrue(_runtimeConfigLoader!.TryLoadConfig(TEST_RUNTIME_CONFIG_FILE, out RuntimeConfig? runtimeConfig));
+
+        // Perform assertions on various properties.
+        Assert.IsNotNull(runtimeConfig);
+        Assert.IsNotNull(runtimeConfig.Runtime);
+        Assert.IsNull(runtimeConfig.Runtime.Telemetry);
+
+        string[] addTelemetryArgs = { "add-telemetry", "-c", TEST_RUNTIME_CONFIG_FILE, "--app-insights-enabled", "true", "--app-insights-conn-string", "InstrumentationKey=00000000" };
```

Comment:
These arguments may not be necessary: "--app-insights-enabled", "true" since conn-string is provided...

Reply:
i feel this should be considering scenarios where the developer may want to store the app-insights conn string but not use it for certain cases where it's impacting performance or due to temp issues with AppInsights server.